commit 6b5189eb2000761cd68c3c54c0d03357a9aaec67
Author: Joseph Myers <joseph@codesourcery.com>
Date:   Thu Nov 27 16:02:26 2014 +0000

    Avoid deprecated sigblock in misc/tst-pselect.c.
    
    misc/tst-pselect.c uses the deprecated sigblock interface, resulting
    in "tst-pselect.c:42:3: warning: 'sigblock' is deprecated (declared at
    ../signal/signal.h:189) [-Wdeprecated-declarations]".  The choice of
    sigblock rather than sigprocmask has nothing to do with what this test
    is testing, so this patch changes it to use sigprocmask to avoid the
    warning.
    
    Tested for x86_64.
    
            * misc/tst-pselect.c (do_test): Use sigprocmask instead of
            sigblock.

diff --git a/misc/tst-pselect.c b/misc/tst-pselect.c
index 36bc46da6c94174a..095d794cb2735e18 100644
--- a/misc/tst-pselect.c
+++ b/misc/tst-pselect.c
@@ -39,9 +39,12 @@ do_test (void)
       return 1;
     }
 
-  if (sigblock (sigmask (SIGUSR1)) != 0)
+  sigset_t ss_usr1;
+  sigemptyset (&ss_usr1);
+  sigaddset (&ss_usr1, SIGUSR1);
+  if (sigprocmask (SIG_BLOCK, &ss_usr1, NULL) != 0)
     {
-      puts ("sigblock failed");
+      puts ("sigprocmask failed");
       return 1;
     }
 
